hedyorg/hedy
# [UI] when running the same code twice, green "good job" modal and confetti button no longer appear #1431

**Description**

* **Closes** [UI] When running the same code twice, both the green "good job" modal and confetti button should no longer appear #1431
* Only shows the modal and button on first run/ if the code has changed

**How to test**

* Go to http://localhost:5000/hedy/2
* In the introduction exercise, click 'Run code' button. There should be good job modal and confetti
* Click 'Run code' button again with no changes. There should be no good job modal and confetti
* Make a change to hello world code and rerun. The good job modal and confetti should be back 🎉
